docker/docker_client_test.go

```diff
@@ -88,194 +88,215 @@ func TestDockerCertDir(t *testing.T) {
-			}),
-			expectedUsername: "user",
-			expectedPassword: "pw",
-		},
-		{
-			name:     "match localhost",
-			hostname: "http://localhost",
-			authConfig: makeTestAuthConfig(testAuthConfigDataMap{
-				"docker.io":   testAuthConfigData{"user", "pw"},
-				"localhost":   testAuthConfigData{"joe", "pass"},
-				"example.com": testAuthConfigData{"alice", "pwd"},
-			}),
-			expectedUsername: "joe",
-			expectedPassword: "pass",
-		},
-		{
-			name:     "match ip",
-			hostname: "10.10.3.56:5000",
-			authConfig: makeTestAuthConfig(testAuthConfigDataMap{
-				"10.10.30.45":     testAuthConfigData{"user", "pw"},
-				"localhost":       testAuthConfigData{"joe", "pass"},
-				"10.10.3.56":      testAuthConfigData{"alice", "pwd"},
-				"10.10.3.56:5000": testAuthConfigData{"me", "mine"},
-			}),
-			expectedUsername: "me",
-			expectedPassword: "mine",
-		},
-		{
-			name:     "match port",
-			hostname: "https://localhost:5000",
-			authConfig: makeTestAuthConfig(testAuthConfigDataMap{
-				"https://127.0.0.1:5000": testAuthConfigData{"user", "pw"},
-				"http://localhost":       testAuthConfigData{"joe", "pass"},
-				"https://localhost:5001": testAuthConfigData{"alice", "pwd"},
-				"localhost:5000":         testAuthConfigData{"me", "mine"},
-			}),
-			expectedUsername: "me",
-			expectedPassword: "mine",
-		},
-		{
-			name:     "use system context",
-			hostname: "example.org",
-			authConfig: makeTestAuthConfig(testAuthConfigDataMap{
-				"example.org": testAuthConfigData{"user", "pw"},
-			}),
-			expectedUsername: "foo",
-			expectedPassword: "bar",
-			ctx: &types.SystemContext{
-				DockerAuthConfig: &types.DockerAuthConfig{
-					Username: "foo",
-					Password: "bar",
+	for _, configPath := range configPaths {
+		for _, tc := range []struct {
+			name             string
+			hostname         string
+			authConfig       testAuthConfig
+			expectedUsername string
+			expectedPassword string
+			expectedError    error
+			ctx              *types.SystemContext
+		}{
+			{
+				name:       "empty hostname",
+				authConfig: makeTestAuthConfig(testAuthConfigDataMap{"localhost:5000": testAuthConfigData{"bob", "password"}}),
+			},
+			{
+				name:     "no auth config",
+				hostname: "index.docker.io",
+			},
+			{
+				name:             "match one",
+				hostname:         "example.org",
+				authConfig:       makeTestAuthConfig(testAuthConfigDataMap{"example.org": testAuthConfigData{"joe", "mypass"}}),
+				expectedUsername: "joe",
+				expectedPassword: "mypass",
+			},
+			{
+				name:       "match none",
+				hostname:   "registry.example.org",
+				authConfig: makeTestAuthConfig(testAuthConfigDataMap{"example.org": testAuthConfigData{"joe", "mypass"}}),
+			},
+			{
+				name:     "match docker.io",
+				hostname: "docker.io",
+				authConfig: makeTestAuthConfig(testAuthConfigDataMap{
+					"example.org": testAuthConfigData{"example", "org"},
+					"docker.io":   testAuthConfigData{"docker", "io"},
```

Comment:
This test case is now missing the `index.docker.io` key. Is that intentional?

Reply:
my bad, I didn't mean to delete that. Will add it back.

Reply:
fixed